### PR TITLE
Fix linting issues in feature-product and main-product sections

### LINKED ISSUE #1540: The add to cart button is getting a primary class if it's linked to a subscription
**Describe the current behavior**

The add to cart is receiving the `button--primary` class if the dynamic checkout is disabled or if there is a subscription option on that product. 

**Describe the expected behavior**

Not sure if that should be the case 🤔  Shouldn't it get the outline class anyway as long as dynamic checkout is enabled?

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 4.0.0

**Possible solution**

remove the `and` part of this statement: 

**Additional context/screenshots**
It can be seen/tested on this product:

### PR DESCRIPTION
**Pull request by metamoni**
**PR Summary:** 
Linting fixes + Update "Add to cart" button styling to `button--secondary` when dynamic checkout/subscriptions enabled. The button styling fix has been separately reviewed and approved [in this PR](https://github.com/Shopify/dawn/pull/1875).

**Why are these changes introduced?**
In order to fix #1540 we need to make some updates in the `feature-product` and `main-product` sections. This PR includes both linting fixes, as well as the fix for #1540, which has been approved as part of [this PR](https://github.com/Shopify/dawn/pull/1875).

**What approach did you take?**
- formatted with [prettier VS code plugin](https://vault.shopify.io/pages/18759-Code-editor-VS-Code#prettier-liquid-plugin)
- updated "Add to cart" button styling to `button--secondary` when dynamic checkout/subscriptions enabled

**Testing steps/scenarios**
- main product functionality and style should not be affected
- feature product functionality and style should not be affected

- [x] steps in https://github.com/Shopify/dawn/pull/1875 already covered

**Demo links**
- [Store](https://os2-demo.myshopify.com/?preview_theme_id=128403374102)
- [Editor](https://os2-demo.myshopify.com/admin/themes/128403374102/editor)

**Checklist**
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Notified the [help.shopify.com](https://help.shopify.com) team about updates to theme settings (current contact: Kimli)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by KaichenWang** [comment]
For the "demo links" in the PR description, it would be best to link to a theme that is connected to this Github branch (with one-way sync enabled). Right now the link points to your local development theme

Screen Shot 2022-07-25 at 7 16 33 PM

**Review by andrewetchen — COMMENTED** [review]
Thanks for linting these, Monica! Both sections looks good to me.

Before I approve I was wondering if we should take care of this now (in this PR) or whether we should wait until a fix for formatting Liquid markup is added to the prettier-plugin-liquid and take care of it then in another PR? Thoughts? 🤔 

- https://github.com/Shopify/prettier-plugin-liquid/issues/5
- https://github.com/Shopify/prettier-plugin-liquid/discussions/23

The relative lines in `featured-product.liquid` and `main-product.liquid` sections:

**Featured product**

- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/featured-product.liquid#L62
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/featured-product.liquid#L74
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/featured-product.liquid#L120
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/featured-product.liquid#L476
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/featured-product.liquid#L489

**Main product**
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/main-product.liquid#L94
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/main-product.liquid#L110
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/main-product.liquid#L325
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/main-product.liquid#L357
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/main-product.liquid#L732
- https://github.com/Shopify/dawn/blob/fb218bf50374878bbae86a139691c900082bad5e/sections/main-product.liquid#L745

---

Taking a look at https://github.com/Shopify/prettier-plugin-liquid/discussions/23, my vote would be **Option 1. Indent**:

```liquid
{% # assume printWidth: 1 (aka forceBreak) %}
{% 
  render 'foo'
    with arg
    for products,
    foo: bar
%}
```

**Review by KaichenWang — APPROVED** [review]
@andrewetchen Thanks for flagging the updates to `prettier-plugin-liquid`. Happy to see the prettier plugin's features continue to evolve/improve, but I feel we can still go ahead and merge this PR as-is. Do let us know if you or anyone else on the team is working on the same files and are concerned about conflicts. We can definitely also hold off on merging this if needed.

**Pull request merged**
